pkg/alertmanager/alertmanager_metrics.go

```diff
@@ -77,7 +77,7 @@ type alertmanagerMetrics struct {
 
 func newAlertmanagerMetrics() *alertmanagerMetrics {
 	return &alertmanagerMetrics{
-		regs: util.NewUserRegistries(),
+		regs: dskit_metrics.NewTenantRegistries(util_log.Logger),
```

Comment:
I opted for the approach of injecting the logger directly here (This pattern is used in 3 additional places) but can technically be injected from the top-level constructor e.g. `newAlertmanagerMetrics`.

This seems OK IMO, but happy to hear otherwise if there's an extra pattern we're following now.

Reply:
Seems like it should be injected from the top-level constructor based on the (currently) failing lint.